src/buffer_line.rs

```diff
@@ -87,7 +89,26 @@ impl BufferLine {
     pub fn set_wrap(&mut self, wrap: Wrap) -> bool {
         if wrap != self.wrap {
             self.wrap = wrap;
-            self.reset();
+            self.reset_layout();
+            true
+        } else {
+            false
+        }
+    }
+
+    /// Get the Text alignment
+    pub fn align(&self) -> Option<Align> {
+        self.align
+    }
+
+    /// Set the text alignment
+    ///
+    /// Will reset shape and layout if it differs from current alignment.
+    /// Returns true if the line was reset
+    pub fn set_align(&mut self, align: Align) -> bool {
```

Comment:
I would recommend that either set_align takes an Option<Align>, or that the Align enum has a Default variant.

Reply:
I'll change this function signature to accept an option. I intentionally didn't do it because I couldn't think of an equivalent of "Default" or "None" as a UI element.